src/components/Note/Note.tsx

```diff
@@ -32,47 +32,19 @@ export const Note = (props: NoteProps) => {
   const isShared = useAppSelector((state) => state.board.data?.sharedNote === props.noteId);
   const allowStacking = useAppSelector((state) => state.board.data?.allowStacking ?? true);
   const showAuthors = useAppSelector((state) => !!state.board.data?.showAuthors);
+  const me = useAppSelector((state) => state.participants?.self);
   const moderating = useAppSelector((state) => state.view.moderating);
   const isModerator = props.viewer.role === "MODERATOR" || props.viewer.role === "OWNER";
 
   // all authors of a note, including its children if it's a stack.
-  // next to the Participant object there's also helper properties (displayName, isSelf) for easier identification.
-  const authors: ParticipantExtendedInfo[] = useAppSelector((state) => {
+  const authors = useAppSelector((state) => {
     const noteAuthor = state.participants?.others.find((p) => p.user.id === note?.author) ?? state.participants?.self;
     const childrenNoteAuthors = state.notes
       // get all notes which are in the same stack as the main note
       .filter((n) => n.position.stack === props.noteId)
       // find the corresponding author for the respective note in the list of other participants. if none is found, the author is therefore yourself
       .map((c) => state.participants?.others.find((p) => p.user.id === c.author) ?? state.participants?.self);
-    const allAuthorsRaw = [noteAuthor, ...childrenNoteAuthors];
-    // process and filter
-    const allAuthors = allAuthorsRaw
-      .map((a) => {
-        const isSelf = a?.user.id === state.participants?.self.user.id;
-        const displayName = isSelf ? t("Note.me") : a!.user.name;
-        return {
-          ...a,
-          displayName,
-          isSelf,
-        } as ParticipantExtendedInfo;
-      })
-      // remove duplicates (because notes can have multiple children by the same authors)
-      .filter((v, i, self) => self.findIndex((a) => a.user?.id === v.user?.id) === i);
-
-    // if self is part of the authors, we always want it to be visible
-    const selfIndex = allAuthors.findIndex((a) => a.isSelf);
-    if (selfIndex > 1) {
-      // in-place swap with second author
-      [allAuthors[selfIndex], allAuthors[1]] = [allAuthors[1], allAuthors[selfIndex]];
-    }
-
-    // if showAuthors is disabled, we still want to see cards written by yourself if you're the stack author.
-    // the other authors are excluded as we only require the stack author
-    if (!showAuthors && props.viewer.user.id === noteAuthor!.user!.id) {
-      return [allAuthors[0]]; // stack author is always first element
-    }
-
-    return allAuthors;
+    return [noteAuthor, ...childrenNoteAuthors].filter((a): a is Participant => !!a); // remove undefined values (which shouldn't exists, only for TS type assertion)
```

Comment:
```suggestion
    return [noteAuthor, ...childrenNoteAuthors].filter(Boolean) as Participant[]; // remove undefined values (which shouldn't exists, only for TS type assertion)
```

This is more readable imo, what do you think?